components/camel-ipfs/src/main/java/org/apache/camel/component/ipfs/IPFSEndpoint.java

```diff
@@ -117,6 +117,8 @@ Path ipfsGet(String cid, Path outdir) throws IOException, TimeoutException {
     }
 
     private IPFSClient ipfs() {
-        return getComponent().getIPFSClient();
+        IPFSClient client = getComponent().getIPFSClient();
```

Comment:
Okay this is better where it can self-heal and reconnect. 

Fix checkstyle issue however!